### ISSUE TITLE
csv: limit how much data is downloaded for the preview

### ISSUE THREAD

**Issue by slint** [inveniosoftware/invenio-previewer#113]
In cases where the delimiter is wrongly detected or possibly, the rows are too big, the previewer tries to fetch too much data from the file, which of course is terrible if someone is on a mobile data connection. To solve this a (configurable) data limit has to be applied to the JavaScript that fetches the initial rows from the CSV file.

See for example the previewer trying to work with a 1.2GB CSV file: https://zenodo.org/record/3731542

**Comment by tmorrell** [comment]
Should have been resolved with https://github.com/inveniosoftware/invenio-previewer/pull/171

**Comment by slint** [comment]
> Should have been resolved with #171

Correct! But it was done in the `maint-1.0.0alpha` branch, so I opened a "forward-port" (?) PR at https://github.com/inveniosoftware/invenio-previewer/pull/185.

The solution differs from the one I described in the initial issue since it applies a strict threshold as a validation check for previewing the file (but is good enough for now). 

The best would be to implement this on the client side, but we're already in discussions about if/how we keep the current D3-based previewer (in case there's a better one out in the wild :) )
